fiftyone/core/odm/database.py

```diff
@@ -126,13 +126,13 @@ def _validate_db_version(config, client):
         # that the user has set their feature compatibility version
         # https://docs.mongodb.com/manual/reference/command/setFeatureCompatibilityVersion
         if version < min_ver:
-            raise RuntimeError(
-                "Found `mongod` version %s, but %s or later is required"
+            logger.warning(
```

Comment:
I hate to go too architecture astronaut-y here, but if we're going to allow this, then perhaps we should also remove the `database_uri` conditional and also provide a config setting that users who are intentionally violating this can set to avoid seeing this warning

Reply:
Ok, I've added the `database_validation` setting and simplified things to one exception. I've changed it back to a `RuntimeError` so a user really has to understand what they are doing.